tests/interfaces/default_qubit_2_integration/test_autograd_default_qubit_2.py

```diff
@@ -123,11 +118,35 @@ def f(x):
 # add tests for lightning 2 when possible
 # set rng for device when possible
 test_matrix = [
-    ({"gradient_fn": param_shift}, 100000, DefaultQubit(seed=42)),
-    ({"gradient_fn": param_shift}, None, DefaultQubit()),
-    ({"gradient_fn": "backprop"}, None, DefaultQubit()),
-    ({"gradient_fn": "adjoint", "grad_on_execution": True}, None, DefaultQubit()),
-    ({"gradient_fn": "adjoint", "grad_on_execution": False}, None, DefaultQubit()),
+    ({"gradient_fn": param_shift}, Shots(100000), DefaultQubit(seed=42)),
+    ({"gradient_fn": param_shift}, Shots((100000, 100000)), DefaultQubit(seed=42)),
+    ({"gradient_fn": param_shift}, Shots(None), DefaultQubit()),
+    ({"gradient_fn": "backprop"}, Shots(None), DefaultQubit()),
+    (
+        {"gradient_fn": "adjoint", "grad_on_execution": True, "use_device_jacobian_product": False},
+        Shots(None),
+        DefaultQubit(),
+    ),
+    (
+        {
+            "gradient_fn": "adjoint",
+            "grad_on_execution": False,
+            "use_device_jacobian_product": False,
+        },
+        Shots(None),
+        DefaultQubit(),
+    ),
+    ({"gradient_fn": "adjoint", "use_device_jacobian_product": True}, Shots(None), DefaultQubit()),
+    (
+        {"gradient_fn": "device", "use_device_jacobian_product": False},
+        Shots((100000, 100000)),
+        ParamShiftDerivativesDevice(),
```

Comment:
Does this device need a seed? :thinking:

Reply:
set.